sphinx/writers/latex.py

```diff
@@ -2003,7 +2003,7 @@ def visit_index(self, node, scre=re.compile(r';\s*')):
                     self.body.append(r'\index{%s|see{%s}}' % (p1, p2))
                 else:
                     logger.warning('unknown index entry type %s found', type)
-                self.body.append('%\n')
+                self.body.append('\\ignorespaces ')
```

Comment:
It could be made conditional

                if not node.get('inline', True):
                    self.body.append('\\ignorespaces ')

to limit a bit the number of `\ignorespaces ` added to latex file.

An alternative would be for Sphinx LaTeX style file to redefine `\index` to always do the `\ignorespaces `, so that the latter would be hidden in redefined `\index` macro. As there is also issue #4071 related to `index`, final decision can be postponed.

Reply:
ah sorry, I think the

        if not node.get('inline', True):
            self.body.append('\\ignorespaces ')

could be moved out of the `for` loop and inserted right before the final

    raise nodes.SkipNode

this will reduce the number of  `\ignorespaces ` added to latex file. And in case of error with the index key, the added `\ignorespaces ` causes no harm anyhow in latex output.